RELEASE_NOTES.md

```diff
@@ -2,6 +2,21 @@
 
 This section contains past updates we've sent to customers via Discord.
 
+## 0.7.0 (June 28, 2023)
+
+We've just released Shorebird v0.7.0 🎉
+
+- 🚀 introduce new free hobby tier
```

Comment:
Let's "launch" the hobby tier tomorrow with the website and remove it from mention in this PR for now.  otherwise lgtm.